ksql-engine/src/main/java/io/confluent/ksql/physical/PhysicalPlanBuilder.java

```diff
@@ -435,9 +437,20 @@ private static void throwIncompatibleKeysException(
   }
 
   // Package private because of test
-  String getServiceId() {
-    return KsqlConstants.KSQL_INTERNAL_TOPIC_PREFIX
-           + ksqlConfig.getString(KsqlConfig.KSQL_SERVICE_ID_CONFIG);
+  String getServiceId(final ServiceContext serviceContext) {
```

Comment:
I feel like the name of this method is now misleading - we aren't actually getting the serviceId anymore! I would split this into two methods: `getServiceId` and `appendUsername` and only call `appendUsername` on transient queries to make sure that we aren't accidentally appending a username when we create the persistent query

Reply:
True. I missed that I am doing the same for persistent queries.
Done.